app/questionnaire/questionnaire_schema.py

```diff
@@ -248,11 +248,8 @@ def _get_answers_by_id(self) -> dict[str, list[ImmutableDict]]:
 
     def _populate_answer_dependencies(self) -> None:
         for block in self.get_blocks():
-            if block["type"] == "CalculatedSummary":
-                answer_ids_for_block = get_calculated_summary_answer_ids(block)
-                self._update_answer_dependencies_for_calculated_summary(
-                    answer_ids_for_block, block["id"]
-                )
+            if block["type"] in {"CalculatedSummary", "GrandCalculatedSummary"}:
```

Comment:
Here we should reuse already existing `is_summary_with_calculation()`  passing block["type"] as an argument.

Reply:
moved the method into questionnaire_schema and imported it in the filters so it can be used here too